src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -216,18 +218,40 @@ export const addCustomElementInputs = (
       // correct virtual module, if we instead referenced, for instance,
       // `cmp.sourceFilePath`, we would end up with duplicated modules in our
       // output.
-      indexImports.push(
+      indexExports.push(
         `export { ${exportName}, defineCustomElement as defineCustomElement${exportName} } from '${coreKey}';`
       );
     }
 
+    indexImports.push(`import { ${exportName} } from '${coreKey}';`);
+    exportNames.push(exportName);
+
     bundleOpts.inputs[cmp.tagName] = coreKey;
     bundleOpts.loader![coreKey] = exp.join('\n');
   });
 
+  if (outputTarget.customElementsExportBehavior === 'bundle') {
+    bundleOpts.loader!['\0core'] += indexImports.join('\n');
+  }
+
   // Only re-export component definitions if the barrel export behavior is set
   if (outputTarget.customElementsExportBehavior === 'single-export-module') {
-    bundleOpts.loader!['\0core'] += indexImports.join('\n');
+    bundleOpts.loader!['\0core'] += indexExports.join('\n');
+  }
+
+  if (outputTarget.customElementsExportBehavior === 'bundle') {
```

Comment:
is there any reason why we wouldn't want to merge this into the other block checking the value of `outputTarget` on line 233? just thinking it would be good to keep the behavior specific to each `customElementsExportBehavior` bundled together as much as we can

Reply:
might be good also to hang just a brief explanatory comment above line 233, along the olines of line 237

Reply:
@alicewriteswrongs We probably can. I have them separate for now to try and keep all the import/export declarations at the top of the file and everything else in the "body". I'll see if there is a way to restructure how the file gets written to make this easier.

Reply:
@rwaskiewicz @alicewriteswrongs I pushed a commit that moves some of this stuff around a bit and changes how the entry-point gets generated. Changes the order of the output slightly, but, functionally, all the same. It just makes it so the output file has separate imports, exports, and body segments so that can stay more consistent as the output target behavior inevitably changes in the future!